.github/workflows/publish.yaml

```diff
@@ -0,0 +1,124 @@
+name: docker_publish
+
+on:
+  push:
+    branches:
+    - 'main'
+    - 'dev'
+    tags:
+    - '*'
+
+jobs:
+  lint_test:
+    uses: babylonlabs-io/.github/.github/workflows/reusable_node_lint_test.yml@hiep/add-nodejs-workflows
+    with:
+      run-build: true
+      run-unit-tests: true
+  
+  docker_build:
+    needs: [lint_test]
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        environment: [devnet, staging, testnet, mainnet-private, mainnet]
```

Comment:
This seems to go against the practice we want to achieve in which pipelines regarding our internal deployment environments are decoupled from the service repositories. Don't mind keeping it as is given that we're under time pressure, but let's track finding a solution for this.

Reply:
Noted